portal-impl/src/com/liferay/portal/service/impl/LayoutSetLocalServiceImpl.java

```diff
@@ -333,19 +333,29 @@ public LayoutSet updateLogo(
 		LayoutSetBranch layoutSetBranch = _getLayoutSetBranch(layoutSet);
 
 		if (layoutSetBranch == null) {
-			layoutSet.setModifiedDate(new Date());
-
 			PortalUtil.updateImageId(
```

Comment:
Would it be possible for portalutil to return the latest layoutSet to avoid the following fetch?

Reply:
It's not impossible, but it's a kernel change and would affect many other classes. Also, I'm not entirely sure yet how we could 'reload' an entry in a generic way inside this kernel method.
What is confusing is that we don't explicitly do the overwrite in this method but another query called from here can trigger a hibernate queue flush that "commits" a change of a previously updated record (it doesn't actually do a DB transaction commit, as I can only see it in the DB as uncommitted change, with `SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED ;` in the query).